packages/@aws-cdk/aws-appsync/lib/graphqlapi.ts

```diff
@@ -433,6 +467,165 @@ export class GraphQLApi extends Construct {
     });
   }
 
+  /**
+   * Adds an IAM policy statement associated with this GraphQLApi to an IAM
+   * principal's policy.
+   *
+   * @param grantee The principal (no-op if undefined)
+   * @param resources The set of resources to allow (i.e. ...:[region]:[accountId]:apis/GraphQLId/...)
+   * @param actions The actions that should be granted to the principal (i.e. appsync:graphql )
+   */
+  public grant(grantee: IGrantable, resources: IamResources[], ...actions: string[]): Grant {
```

Comment:
Do we really need to make this public?

The array argument plus the variadic arguments makes me uneasy about this API. Maybe let's leave it at the other `grant*` methods?

If we make this `private`, we can also make `IamResources` package-private, which is an additional bonus.

Reply:
are you suggesting passing in props to configure the permission?

The customer need is the ability to create permissions that allow for exposure to certain graphql resources..

These can be anything from `Query`, `Mutation` or even a custom type that they made 

```gql
type customType @aws_iam {
  ...
}
```

customers have been asking for an easy way to configure `IAM` permissions and the best way seems to be through `grant` because the IAM role must be created and then passed in.

Reply:
Are there any other actions besides `appsync:GraphQL` that can be passed here?

Reply:
yes.. something like `"appsync:CreateFunction"` is allowed and other things like `"appsync:ListDataSource"`

i wanted `grant` to be an escape hatch for interfaces that dont exist yet.

the `grant*` functions were more like a nice to have because almost all apps using graphql will have to interface Query Subscription and Mutation